src/krux/wallet.py

```diff
@@ -58,27 +60,38 @@ def __init__(self, key):
 
     def is_multisig(self):
         """Returns a boolean indicating whether or not the wallet is multisig"""
-        return self.key.multisig
+        if self.key:
+            return self.key.multisig
+        if self.descriptor:
+            return self.descriptor.is_basic_multisig
+        return None
 
     def is_loaded(self):
         """Returns a boolean indicating whether or not this wallet has been loaded"""
         return self.wallet_data is not None
 
     def load(self, wallet_data, qr_format):
         """Loads the wallet from the given data"""
-        descriptor, label = parse_wallet(wallet_data, self.key.network)
-
-        if self.is_multisig():
-            if not descriptor.is_basic_multisig:
-                raise ValueError("not multisig")
-            if self.key.xpub() not in [key.key.to_base58() for key in descriptor.keys]:
-                raise ValueError("xpub not a cosigner")
+        if self.key:
+            network = self.key.network
         else:
-            if not descriptor.key:
-                if len(descriptor.keys) > 1:
-                    raise ValueError("not single-sig")
-            if self.key.xpub() != descriptor.key.key.to_base58():
-                raise ValueError("xpub does not match")
+            network = NETWORKS[Settings().wallet.network]
```

Comment:
What happens if the user customize the wallet to main/test with this line? The Settings only serves for default now, not for the current loaded wallet.

Reply:
I see, this will only applies for when loading only the descriptor right? So why not use the descriptor derivation path info?